include/vkwrap/vkwrap/image.h

```diff
@@ -0,0 +1,302 @@
+#pragma once
+
+#include "common/vulkan_include.h"
+
+#include "vkwrap/core.h"
+#include "vkwrap/mman.h"
+#include "vkwrap/queues.h"
+#include "vkwrap/utils.h"
+
+#include "utils/patchable.h"
+
+#include <range/v3/range/concepts.hpp>
+#include <range/v3/range/conversion.hpp>
+
+#include <exception>
+#include <functional>
+#include <optional>
+#include <utility>
+
+#include <cassert>
+
+namespace vkwrap
+{
+
+namespace detail
+{
+
+inline vk::ImageViewType
+chooseImageViewType( vk::ImageType type )
+{
+    using vk::ImageType;
+    using vk::ImageViewType;
+
+    switch ( type )
+    {
+    case ImageType::e1D:
+        return ImageViewType::e1D;
+
+    case ImageType::e2D:
```

Comment:
Unfortunately, you cannot choose `ImageViewType::e2D` only because you have `ImageType::e2D`. We need `ImageViewType::e2DArray` for texture image view.